[JENKINS-62630] Warnings Plugin Native Format cannot scan console log output

<!-- ### Imported Jira references for easier searching -->
<!-- [jira_issue_key=JENKINS-62630] -->
<!-- [jira_issue_type=Bug] -->
<!-- [reporter=schlegel_m] -->
<!-- [author=schlegel_m] -->
<!-- [jira_component=warnings-ng-plugin] -->
<!-- [jira_label=hacktoberfest] -->
<!-- [jira_label=help-wanted] -->
<!-- [jira_label=newbie-friendly] -->
<!-- [jira_issues_importer_version=1.4.0] -->

<p>Have configured an issue recorder for the native format according to the docs in <a href="https://github.com/jenkinsci/warnings-ng-plugin/blob/master/doc/Documentation.md#export-your-issues-into-a-supported-format" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/warnings-ng-plugin/blob/master/doc/Documentation.md#export-your-issues-into-a-supported-format</a></p>

<pre>
recordIssues(tool: issues()) </pre>
<p>However, the build breaks with the error:</p>


```
java.lang.AssertionError: Static analysis tool Warnings Plugin Native Format cannot scan console log output, please define a file pattern
	at edu.hm.hafner.util.Ensure.throwException(Ensure.java:169)
	at edu.hm.hafner.util.Ensure.access$000(Ensure.java:36)
	at edu.hm.hafner.util.Ensure$BooleanCondition.isTrue(Ensure.java:724)
	at io.jenkins.plugins.analysis.core.model.ReportScanningTool.scanInConsoleLog(ReportScanningTool.java:164)
	at io.jenkins.plugins.analysis.core.model.ReportScanningTool.scan(ReportScanningTool.java:123)
	at io.jenkins.plugins.analysis.core.steps.IssuesScanner.scan(IssuesScanner.java:101)
	at io.jenkins.plugins.analysis.core.steps.IssuesRecorder.scanWithTool(IssuesRecorder.java:666)
	at io.jenkins.plugins.analysis.core.steps.IssuesRecorder.record(IssuesRecorder.java:631)
	at io.jenkins.plugins.analysis.core.steps.IssuesRecorder.perform(IssuesRecorder.java:602)
	at io.jenkins.plugins.analysis.core.steps.RecordIssuesStep$Execution.run(RecordIssuesStep.java:1007)
	at io.jenkins.plugins.analysis.core.steps.RecordIssuesStep$Execution.run(RecordIssuesStep.java:969)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```
<p>Thats not what I would have expected form the docs. </p>

<p>The console log contains entries like these:</p>


```
[testReportI18n] Jun 09, 2020 10:48:44 AM Log
[testReportI18n] SEVERE: {"fileName":"D:\Jenkins\workspace\BI-Reports_General_Reports_trunk\report\Relation Matrix\Relation_Matrix.jrxml","severity":"ERROR","lineStart":1,"description":"(en) missing key 'relation.matrix.object'"}

```
<p>Is this a problem?</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/schlegel_m.jpg" title="schlegel_m's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=schlegel_m">schlegel_m</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-62630" target="_blank">Warnings Plugin Native Format cannot scan console log output</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: warnings-ng-plugin
<li><b>label(s)</b>: hacktoberfest, help-wanted, newbie-friendly
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-09
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Have configured an issue recorder for the native format according to the docs in <a href="https://github.com/jenkinsci/warnings-ng-plugin/blob/master/doc/Documentation.md#export-your-issues-into-a-supported-format" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/warnings-ng-plugin/blob/master/doc/Documentation.md#export-your-issues-into-a-supported-format</a></p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">recordIssues(tool: issues()) </pre>
</div></div>
<p>However, the build breaks with the error:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>java.lang.AssertionError: Static analysis tool Warnings Plugin Native Format cannot scan console log output, please define a file pattern
	at edu.hm.hafner.util.Ensure.throwException(Ensure.java:169)
	at edu.hm.hafner.util.Ensure.access$000(Ensure.java:36)
	at edu.hm.hafner.util.Ensure$BooleanCondition.isTrue(Ensure.java:724)
	at io.jenkins.plugins.analysis.core.model.ReportScanningTool.scanInConsoleLog(ReportScanningTool.java:164)
	at io.jenkins.plugins.analysis.core.model.ReportScanningTool.scan(ReportScanningTool.java:123)
	at io.jenkins.plugins.analysis.core.steps.IssuesScanner.scan(IssuesScanner.java:101)
	at io.jenkins.plugins.analysis.core.steps.IssuesRecorder.scanWithTool(IssuesRecorder.java:666)
	at io.jenkins.plugins.analysis.core.steps.IssuesRecorder.record(IssuesRecorder.java:631)
	at io.jenkins.plugins.analysis.core.steps.IssuesRecorder.perform(IssuesRecorder.java:602)
	at io.jenkins.plugins.analysis.core.steps.RecordIssuesStep$Execution.run(RecordIssuesStep.java:1007)
	at io.jenkins.plugins.analysis.core.steps.RecordIssuesStep$Execution.run(RecordIssuesStep.java:969)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)</pre>
</div></div>
<p>Thats not what I would have expected form the docs. </p>

<p>The console log contains entries like these:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>[testReportI18n] Jun 09, 2020 10:48:44 AM Log
[testReportI18n] SEVERE: {"fileName":"D:\Jenkins\workspace\BI-Reports_General_Reports_trunk\report\Relation Matrix\Relation_Matrix.jrxml","severity":"ERROR","lineStart":1,"description":"(en) missing key 'relation.matrix.object'"}
</pre>
</div></div>
<p>Is this a problem?</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Jenkins 2.238, warnings-ng 8.1.0</code></li></ul>

--- a/doc/Documentation.md
+++ b/doc/Documentation.md
@@ -203,8 +203,10 @@ issues in a simple log file that contains single lines of JSON issues, see
 Here is an example step that can be used to parse the native JSON (or XML) format:
 
 ```groovy
-recordIssues(tool: issues())
+recordIssues(tool: issues(pattern: '**/issues.json'))
 ```
+
+Note: The native format parser requires a file pattern and cannot scan console log output directly.
 #### Deploying a new tool using a custom plugin
 
 The most flexible way is to define a new tool by writing a Java class that will be deployed in your own small 

